src/overlays/actors/ovl_En_Vb_Ball/z_en_vb_ball.c

```diff
@@ -27,17 +32,268 @@ const ActorInit En_Vb_Ball_InitVars = {
+            angle = Math_atan2f(this->actor.posRot.pos.x, this->actor.posRot.pos.z);
+            this->actor.velocity.x = sinf(angle) * 10.0f;
+            this->actor.velocity.z = cosf(angle) * 10.0f;
+            this->actor.velocity.y = this->actor.velocity.y * -0.5f;
+            if (this->actor.params & 1) {
+                Audio_PlaySoundGeneral(NA_SE_EN_VALVAISA_LAND, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                                       &D_801333E8);
+            }
+            for (i = 0; i < 10; i++) {
+                Vec3f dustVel = { 0.0f, 0.0f, 0.0f };
+                Vec3f dustAcc = { 0.0f, 0.0f, 0.0f };
+                Vec3f dustPos;
+                dustVel.x = Math_Rand_CenteredFloat(8.0f);
+                dustVel.y = Math_Rand_ZeroFloat(1.0f);
+                dustVel.z = Math_Rand_CenteredFloat(8.0f);
+                dustAcc.y = 0.3f;
+                dustPos.x = Math_Rand_CenteredFloat(20.0f) + this->actor.posRot.pos.x;
+                dustPos.y = this->actor.groundY + 10.0f;
+                dustPos.z = Math_Rand_CenteredFloat(20.0f) + this->actor.posRot.pos.z;
+                EnVbBall_SpawnDust(globalCtx, bossFd->particles, &dustPos, &dustVel, &dustAcc,
+                                   Math_Rand_ZeroFloat(80.0f) + 200.0f);
+            }
+        }
+    }
+    if (this->actor.posRot.pos.y < 50.0f) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnVbBall_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnVbBall* this = THIS;
+    EnVbBall* newActor;
+    BossFd* bossFd = BOSSFD;
+    f32 pad3;
+    f32 pad2;
+    s16 spawnNum;
+    s16 i;
+    Vec3f spawnOffset;
+    f32 temp_f0;
+    f32 temp_f20;
+
+    this->unkTimer2++;
+    DECR(this->unkTimer1);
+    this->actor.shape.rot.x += (s16)this->xRotVel;
+    this->actor.shape.rot.y += (s16)this->yRotVel;
+    this->actor.velocity.y += -1.0f;
+    this->actor.gravity = -1.0f;
+    func_8002D7EC(&this->actor);
+    if (this->actor.params >= 0xC8) {
+        EnVbBall_UpdateBones(this, globalCtx);
+    } else {
+        Math_SmoothScaleMaxF(&this->shadowOpacity, 175.0f, 1.0f, 40.0f);
+        temp_f20 = this->actor.scale.y * 1700.0f;
+        this->actor.posRot.pos.y -= temp_f20;
+        func_8002E4B4(globalCtx, &this->actor, 50.0f, 50.0f, 100.0f, 4);
+        this->actor.posRot.pos.y += temp_f20;
+        if ((this->actor.bgCheckFlags & 1) && (this->actor.velocity.y <= 0.0f)) {
+            if ((this->actor.params == 0x64) || (this->actor.params == 0x65)) {
+                Actor_Kill(&this->actor);
+                if (this->actor.params == 0x64) {
+                    func_80033E88(&this->actor, globalCtx, 5, 0xA);
+                }
+                if (this->actor.params == 0x64) {
+                    spawnNum = 2;
+                } else {
+                    spawnNum = 2;
+                }
+                for (i = 0; i < spawnNum; i++) {
+                    if (this->actor.params == 0x64) {
```

Comment:
So I know in general we keep hex for the actor variable but this one was clearly intended as decimal, it compares to 0xC8=200, 0x64=100 and is used in at least one additon
Would it be ok to not use hex for this one then?